libweb3jsonrpc/UnixSocketServer.cpp

```diff
@@ -111,14 +111,15 @@ void UnixDomainSocketServer::Listen()
 			DEV_GUARDED(x_sockets)
 				m_sockets.insert(connection);
 
-			std::thread handler([this, connection](){ GenerateResponse(connection); });
+			std::thread handler([this, connection](){ GenerateResponse(connection); CloseConnection(connection);});
 			handler.detach();
 		}
 	}
 }
 
 void UnixDomainSocketServer::CloseConnection(int _socket)
 {
+	shutdown(_socket, SHUT_RDWR);
```

Comment:
I don't understand why this is needed. Seems to be redundant to `close()`.

In any case, can you also handle error codes?